include/dslash.h

```diff
@@ -327,8 +329,9 @@ namespace quda
 
     Arg &dslashParam; // temporary addition for policy compatibility
 
-    Dslash(Arg &arg, const ColorSpinorField &out, const ColorSpinorField &in, const std::string &app_base = "") :
-      TunableKernel3D(in, 1, arg.nParity), arg(arg), out(out), in(in), nDimComms(4), dslashParam(arg)
+    Dslash(Arg &arg, cvector_ref<ColorSpinorField> &out, cvector_ref<const ColorSpinorField> &in,
+           const ColorSpinorField &halo, const std::string &app_base = "") :
+      TunableKernel3D(in[0], halo.X(4), arg.nParity), arg(arg), out(out), in(in), halo(halo), nDimComms(4), dslashParam(arg)
```

Comment:
Should we add a version of `TunableKernel3D` (and others) that can take in `vector_ref`? I can see there being a case for avoiding this one, just because (iirc) the tunable kernel classes take in a `LatticeField&`, so this would open up a whole other can of worms.

Reply:
Yeah, I would skip that for now.